### ISSUE TITLE
numeric-separators-style: option to accept numbers without grouping

### ISSUE THREAD

**Issue by kevinoid** [sindresorhus/eslint-plugin-unicorn#868]
I'm a big fan of the new [`numeric-separators-style`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/numeric-separators-style.md) rule from #833.  Great idea!

I'd like to enable the `numeric-separators-style` rule to catch incorrect groupings without requiring that all numbers use separators.  For example:
```js
const foo = 1234;  // Pass: No separators
const bar = 1_234; // Pass: Correct grouping
const baz = 12_34; // Fail: Incorrect grouping
```
I couldn't find a way to achieve this.  Configuring `minimum` between 0 and 4, `foo` and `baz` fail.  With `minimum` 5 or above, `bar` and `baz` fail.

**Is this an feature that you might consider?  If so, how to configure?**

An idea:  Additional options for each number type:

| Name | Description |
| --- | --- |
| `requireLeft` | Require separators to the left of the decimal point for numbers with at least `minimum` digits to the left.  (Current behavior: `true`) |
| `requireRight` | Require separators to the right of the decimal point for numbers with at least `minimum` digits to the right.  (Current behavior: `true`) |
| `forbidLeft` | Forbid separators to the left of the decimal point for numbers with less than `minimum` digits to the left.  (Current behavior: `true`) |
| `forbidRight` | Forbid separators to the right of the decimal point for numbers with less than `minimum` digits to the right.  (Current behavior: `true`) |

The desired example behavior could then be accomplished with `requireLeft: false` (to never require grouping) or something like `minimum: 7, forbidLeft: false` to require grouping 7-digits or more, but allow grouping fewer.

**Comment by kevinoid** [comment]
@noftaly Thanks for proposing and implementing this rule!  Any thoughts on this feature?

**Comment by fisker** [comment]
How about just `number: "ignore"` to ignore it.

**Comment by kevinoid** [comment]
> How about just `number: "ignore"` to ignore it.

Wouldn't that pass `foo`, `bar`, and `baz` in my original example?  I'd like to fail `baz`.

**Comment by noftaly** [comment]
> I'd like to enable the numeric-separators-style rule to catch incorrect groupings without requiring that all numbers use separators.

If i understand correctly, you want to only catch incorrect grouping if the number contains numeric separators? Like
```js
const foo = 100000; // Pass
const bar = 1_000_000; // Pass
const baz = 10_0_000_0; // Fail
```
Or to do the above but only if there are less than the specified? `minimumDigits`

If yes, then I think this is a great idea 🙂 I don't really have the time to work on the right now (I'll be able to in a couple of weeks), so if someone wants to do it in the meantime, go ahead!

**Comment by kevinoid** [comment]
> If i understand correctly, you want to only catch incorrect grouping if the number contains numeric separators?

Yes, exactly.

> Or to do the above but only if there are less than the specified?

I could imagine use cases/style guides which require grouping for numbers that exceed a certain size, so tried to cover that in my proposal.  I don't have a current need, so if the accepted solution doesn't address it, that's fine with me

> `minimumDigits`

Were you referring to the existing `minimum` configuration option or proposing a new `minimumDigits` option to address this issue?  Could you clarify how `minimumDigits` would interact with `minimum`?

**Comment by noftaly** [comment]
> I could imagine use cases/style guides which require grouping for numbers that exceed a certain size, so tried to cover that in my proposal.

Why not, but I think we should not add a ton of options for only one single rule, and keep the configuration clear... I'll let the core team discuss on that 😄 


> Were you referring to the existing minimum configuration option or proposing a new minimumDigits option to address this issue? Could you clarify how minimumDigits would interact with minimum?

Yes sorry for the confusion, I was speaking about the `minimum` option. I will edit my comment to avoid confusing other peoples 🙂

**Comment by fisker** [comment]
To be honest, I don't see why you want `const foo = 1234;` pass? That's the point of this rule.

**Comment by kevinoid** [comment]
> To be honest, I don't see why you want const foo = 1234; pass? That's the point of this rule.

Because sometimes it's desirable to catch incorrect/misleading grouping, but not to force minor style details everywhere?  Regardless of the rationale, this rule already has the `minimumDigits` configuration option to support it.  Part of the problem is that `minimumDigits: 4` would make `const foo = 1_234;` invalid.

P.S.  @noftaly I had `minimum` and `minimumDigits` confused.  You had the right name all along.  My bad.
